modules/common-fs2/src/test/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/blackbox/BlackBoxTesting.scala

```diff
@@ -0,0 +1,135 @@
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.common.fs2.blackbox
+
+import scala.collection.JavaConverters._
+
+import org.specs2.mutable.Specification
+
+import cats.effect.IO
+
+import cats.effect.testing.specs2.CatsIO
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+
+import io.circe.Json
+import io.circe.literal._
+import io.circe.syntax._
+
+import org.apache.thrift.TSerializer
+
+import com.snowplowanalytics.snowplow.CollectorPayload.thrift.model1.CollectorPayload
+
+import com.snowplowanalytics.iglu.client.{CirceValidator, Client, Resolver}
+import com.snowplowanalytics.iglu.client.resolver.registries.Registry
+
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaVer, SelfDescribingData}
+import com.snowplowanalytics.iglu.core.circe.implicits._
+
+import com.snowplowanalytics.snowplow.enrich.common.utils.BlockerF
+import com.snowplowanalytics.snowplow.enrich.common.outputs.EnrichedEvent
+import com.snowplowanalytics.snowplow.enrich.common.enrichments.EnrichmentRegistry
+
+import com.snowplowanalytics.snowplow.enrich.common.fs2.Enrich
+
+import com.snowplowanalytics.snowplow.enrich.common.fs2.EnrichSpec
+
+object BlackBoxTesting extends Specification with CatsIO {
+
+  val igluClient: Client[IO, Json] =
+    Client[IO, Json](Resolver(List(Registry.EmbeddedRegistry), None), CirceValidator)
+
+  private val serializer: TSerializer = new TSerializer()
+
+  def buildCollectorPayload(
+    body: Option[String] = None,
+    contentType: Option[String] = None,
+    headers: List[String] = Nil,
+    ipAddress: String = "",
+    networkUserId: String = java.util.UUID.randomUUID().toString,
+    path: String = "",
+    querystring: Option[String] = None,
+    refererUri: Option[String] = None,
+    timestamp: Long = 0L,
+    userAgent: Option[String] = None
+  ): Array[Byte] = {
+    val cp = new CollectorPayload(
+      "iglu:com.snowplowanalytics.snowplow/CollectorPayload/thrift/1-0-0",
+      ipAddress,
+      timestamp,
+      "UTF-8",
+      "ssc"
+    )
+    cp.body = body.orNull
+    cp.contentType = contentType.orNull
+    cp.hostname = "hostname"
+    cp.headers = headers.asJava
+    cp.ipAddress = ipAddress
+    cp.networkUserId = networkUserId
+    cp.path = path
+    cp.querystring = querystring.orNull
+    cp.refererUri = refererUri.orNull
+    cp.userAgent = userAgent.orNull
+    serializer.serialize(cp)
+  }
+
+  def runTest(
+    input: Array[Byte],
+    expected: Map[String, String],
+    enrichmentConfig: Option[Json] = None
+  ) =
+    Enrich
+      .enrichWith(getEnrichmentRegistry(enrichmentConfig), igluClient, None, EnrichSpec.processor, true)(
+        input
+      )
+      .map {
+        case (List(Validated.Valid(enriched)), _) => checkEnriched(enriched, expected)
+        case other => ko(s"there should be one enriched event but got $other")
```

Comment:
Agreed that it would be great but these black box tests contain all the tests migrated from Beam Enrich and there is no such case, so leaving it as is for now.
But don't worry, this use case of a `CollectorPayload` holding multiple events is tested in other places.